imports/plugins/core/graphql/lib/hocs/withCatalogItems.js

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Query } from "react-apollo";
+import getCatalogItems from "../queries/getCatalogItems";
+
+export default (Component) => (
+  class CatalogItems extends React.Component {
+    static propTypes = {
+      shopId: PropTypes.string,
+      shouldSkipGraphql: PropTypes.bool, // Whether to skip this HOC's GraphQL query & data
+      tagId: PropTypes.string
+    };
+
+    render() {
+      const { shouldSkipGraphql, shopId, tagId } = this.props;
+
+      if (shouldSkipGraphql || !shopId) {
+        return (
+          <Component {...this.props} />
+        );
+      }
+
+      const variables = { shopId };
+
+      if (tagId) {
+        variables.tagIds = [tagId];
+      }
+
+      return (
+        <Query query={getCatalogItems} variables={variables}>
+          {({ loading, data, fetchMore }) => {
+            const props = {
+              ...this.props,
+              isLoadingCatalogItems: loading,
+              fetchMore
+            };
+
+            if (loading === false) {
+              const { catalogItems } = data;
+              props.catalogItems = catalogItems || {};
```

Comment:
I think it's best if we pass just the item nodes, i.e., do the map to `item.node` here before passing down. The wrapped component in almost all cases will not care about edges or cursors. If it ever needs them, we can add an option to provide them.

Also, fetchMore/loadMore should be passed down with `pageInfo` already bound to it. The wrapped component shouldn't need to know anything about `pageInfo`, cursors, or pagination. All it knows is that it calls `this.props.loadMore` when necessary, and you can pass down a `hasMoreItems` boolean so that it knows whether to display a load more button.

This is the pattern we've tried to use in the starter kit. It's a bit more complicated scenario, but you can see what I mean here: https://github.com/reactioncommerce/reaction-next-starterkit/blob/de613530d6aff10bdf5023d910b5ee07c06e127a/src/containers/cart/withCart.js#L205-L206